shopinvader_api_sale/routers/sales.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Akretion (https://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+from typing import Annotated
+
+from fastapi import APIRouter, Depends
+
+from odoo import api, models
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.extendable_fastapi.schemas import PagedCollection
+from odoo.addons.fastapi.dependencies import (
+    authenticated_partner,
+    authenticated_partner_env,
+    paging,
+)
+from odoo.addons.fastapi.schemas import Paging
+from odoo.addons.fastapi.utils import FilteredDomainAdapter
+from odoo.addons.shopinvader_schema_sale.schemas import Sale, SaleSearch
+
+sale_router = APIRouter(tags=["sales"])
+
+
+@sale_router.get("/sales")
+def search(
+    params: Annotated[SaleSearch, Depends()],
+    paging: Annotated[Paging, Depends(paging)],
+    env: Annotated[api.Environment, Depends(authenticated_partner_env)],
+    partner: Annotated["ResPartner", Depends(authenticated_partner)],
+) -> PagedCollection[Sale]:
+    """Get the list of sale orders."""
+    count, orders = env["shopinvader_api_sale.sales_router.helper"]._search(
```

Comment:
@sebastienbeau IMO, the parnter should be explicitly set into the domain.

Reply:
I do not think it's needed as we already have the ir.rule that filter the sale.order.

Reply:
After thinking twice, adding explicit domain is needed as some other module can add extra ir.rule and this can lead to leak data on unwanted endpoint

Here is a proposal with an explicit domain 
https://github.com/akretion/odoo-shopinvader/pull/10